distribution/pull_v2.go

```diff
@@ -356,6 +360,12 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		return false, fmt.Errorf("image manifest does not exist for tag or digest %q", tagOrDigest)
 	}
 
+	if m, ok := manifest.(*schema2.DeserializedManifest); ok {
+		if m.Manifest.Config.MediaType == schema2.MediaTypePluginConfig {
```

Comment:
I believe only `schema2.MediaTypeConfig` is expected value here. Wondering if we could just throw an error of an expected configuration type if that is not seen. Ping @aaronlehmann